server/src/main/java/org/apache/druid/segment/realtime/appenderator/StreamAppenderator.java

```diff
@@ -166,7 +166,16 @@
 
   private volatile Throwable persistError;
 
+
+  /**
+   * Flag to tell internals whether appenderator is working on behalf of a real time task.
+   * This is to manage certain aspects as needed. For example, for batch, non-real time tasks,
+   * physical segments (i.e. hydrants) do not need to memory map their persisted
+   * files. In this case, the code will avoid memory mapping them thus ameliorating the occurance
+   * of OOMs.
+   */
   private final boolean isRealTime;
```

Comment:
it seems like this should just be removed? Since this functionality hasn't been released, should it still be here? I guess it could be removed in a follow-up, but I guess it also means there is no way to turn _this_ functionality off for batch tasks in case closing the segments in `StreamAppenderator` itself has a bug.

Reply:
It is used for the fall back flag.

Reply:
It is not really though, when the fallback is set to true, `StreamAppenderator` is made with isRealtime hard coded to false, instead of controlled by a flag as was introduced in #11123. This means there is no way to revert the behavior of that PR since it isn't operator controllable anymore. Rather than introduce a 2nd flag, I strongly think we should consider removing the isRealtime from `StreamAppenderator` before 0.22 since the behavior was previously unreleased and now there is no way to not use it (I don't think there is enough time to have no setting at all and always use `BatchAppenderator`, so that flag will still need to exist to choose between them).